Tests/Algorithm/AlgorithmAddDataTests.cs

```diff
@@ -393,6 +395,61 @@ public void AddDataSecurityTickerNoUnderlying(string ticker, Type customDataType
             }
         }
 
+        [Test]
+        public void AddOptionWithUnderlyingFuture()
+        {
+            // Adds an option containing a Future as its underlying Symbol.
+            // This is an essential step in enabling custom derivatives
+            // based on any asset class provided to Option. This test
+            // checks the ability to create Future Options.
+            var algo = new QCAlgorithm();
+            algo.SubscriptionManager.SetDataManager(new DataManagerStub(algo));
+
+            var underlying = algo.AddFuture("ES", Resolution.Minute, Market.CME);
+            underlying.SetFilter(0, 365);
+
+            var futureOption = algo.AddOption(underlying.Symbol, Resolution.Minute);
+
+            Assert.IsTrue(futureOption.Symbol.HasUnderlying);
+            Assert.AreEqual(underlying.Symbol, futureOption.Symbol.Underlying);
```

Comment:
Nice! We could add some test to assert the Margin Model being used, not sure where those tests are but they must be around

Reply:
Ah found it, you already added it 😁